docs/en/latest/terminology/plugin.md

```diff
@@ -95,6 +95,43 @@ the configuration above means customizing the error response from the jwt-auth p
 | Name         | Type | Description |
 |--------------|------|-------------|
 | error_response | string/object  | Custom error response |
+| priority       | integer        | Custom plugin priority |
+
+### Custom Plugin Priority
+
+All plugins have a default priority, but it is possible to customize the plugin priority to change the plugin's execution order.
+
+```json
+ {
+    "serverless-post-function": {
+        "_meta": {
+            "priority": 10000
+        },
+        "phase": "rewrite",
+        "functions" : ["return function(conf, ctx)
+                    ngx.say(\"serverless-post-function\");
+                    end"]
+    },
+    "serverless-pre-function": {
+        "_meta": {
+            "priority": -2000
+        },
+        "phase": "rewrite",
+        "functions": ["return function(conf, ctx)
+                    ngx.say(\"serverless-pre-function\");
+                    end"]
+    }
+}
+```
+
+The default priority of serverless-pre-function is 10000, and the default priority of serverless-post-function is -2000. By default, the serverless-pre-function plugin will be executed first, and serverless-post-function plugin will be executed next.
```

Comment:
At this point, why do we keep two different functions? We can keep a single one as call it whenever we required it. Can we change the sample?

Reply:
A good use-case would be the combination of `proxy-mirror` plugin and the `proxy-rewrite` one.

Reply:
> A good use-case would be the combination of `proxy-mirror` plugin and the `proxy-rewrite` one.

I tried it and the two plugins don't show well to work together by customizing the priority.

Maybe you're talking about changing the request path via the `proxy-rewrite` plugin, and then `proxy-mirror` using that new path.

But `proxy-rewrite` plugin update uri on the var `ctx.var.upstream_uri`, but `proxy-mirror` get the uri by the var `ctx.var.uri`, this is the difference in implementation.

Reply:
>I tried it and the two plugins don't show well to work together by customizing the priority.

That's actually the origin of my initial requirement.

>Maybe you're talking about changing the request path via the proxy-rewrite plugin, and then proxy-mirror using that new path.

Indeed!

Imagine that I receive a request for `v1/hello`. Now, my requirements are:

1. The prefix `v1` to be removed
2. The request to be forwarded to the upstream
3. The request to be duplicated to the mirror

>But proxy-rewrite plugin update uri on the var ctx.var.upstream_uri, but proxy-mirror get the uri by the var ctx.var.uri, this is the difference in implementation.

Again, how are users supposed to know this? It's an implementation detail. 

The requirements make sense from a functional point of view.

Reply:
Since `proxy-mirror` support rewrite `path`, so we can implement the above requirements, but we just need to configure them separately on `proxy-mirror` and `proxy-rewrite`.

Of course, we can also modify the `proxy-mirror` plugin to read the `ctx.var.upstream_uri` variable. Of course, it won't be in this PR.

Reply:
> we can also modify the proxy-mirror plugin to read the ctx.var.upstream_uri variable

in this way, we can match the requirements